### ISSUE TITLE
AlreadyExistsError

### ISSUE THREAD

**Issue by shuocaicai** [zipengxuc/ADVSE-GuessWhat#2]
I ran into a problem replicating your code. Here's how:

`Exception has occurred: InvalidArgumentError
indices[0] = 4902 is not in [0, 4902)
	 [[node qgen/embedding_lookup (defined at /data/caishuo/ADVSE_v2/src/guesswhat/models/qgen/qgen_uaqrah_network.py:78) ]]

Errors may have originated from an input operation.
Input Source operations connected to node qgen/embedding_lookup:
 qgen/word_embedding/dialogue_embedding_encoder/read (defined at /data/caishuo/ADVSE_v2/src/guesswhat/models/qgen/qgen_uaqrah_network.py:43)

Original stack trace for 'qgen/embedding_lookup':
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/shuocai/.vscode-server/extensions/ms-python.python-2022.4.0/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/home/shuocai/.vscode-server/extensions/ms-python.python-2022.4.0/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 444, in main
    run()
  File "/home/shuocai/.vscode-server/extensions/ms-python.python-2022.4.0/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 285, in run_file
    runpy.run_path(target_as_str, run_name=compat.force_str("__main__"))
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "src/guesswhat/train/train_qgen_supervised.py", line 98, in <module>
    network, batchifier_cstor = create_qgen(config["model"], num_words=tokenizer.no_words)
  File "/data/caishuo/ADVSE_v2/src/guesswhat/models/qgen/qgen_factory.py", line 10, in create_qgen
    network = QGenNetworkHREDDecoderUAQRAH(config, num_words, rl_module=rl_module)
  File "/data/caishuo/ADVSE_v2/src/guesswhat/models/qgen/qgen_uaqrah_network.py", line 78, in __init__
    q_ini = tf.nn.embedding_lookup(params=self.dialogue_emb_weights, ids=tf.fill([bs], num_words))
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/ops/embedding_ops.py", line 315, in embedding_lookup
    transform_fn=None)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/ops/embedding_ops.py", line 133, in _embedding_lookup_and_transform
    array_ops.gather(params[0], ids, name=name), ids, max_norm)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/ops/array_ops.py", line 3475, in gather
    return gen_array_ops.gather_v2(params, indices, axis, name=name)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/ops/gen_array_ops.py", line 4097, in gather_v2
    batch_dims=batch_dims, name=name)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3616, in create_op
    op_def=op_def)
  File "/home/shuocai/anaconda3/envs/guesswhat/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2005, in __init__
    self._traceback = tf_stack.extract_stack()

During handling of the above exception, another exception occurred:

  File "/data/caishuo/ADVSE_v2/src/generic/tf_utils/evaluator.py", line 96, in execute
    return sess.run(output, feed_dict=feed_dict)
  File "/data/caishuo/ADVSE_v2/src/generic/tf_utils/evaluator.py", line 70, in process
    results = self.execute(sess, outputs, batch)
  File "/data/caishuo/ADVSE_v2/src/guesswhat/train/train_qgen_supervised.py", line 147, in <module>
    [train_loss, _] = evaluator.process(sess, train_iterator, outputs=outputs + [optimizer])`

First I encountered an index error, and I located the position where the error occurred：
`File "/data/caishuo/ADVSE_v2/src/guesswhat/models/qgen/qgen_uaqrah_network.py", line 78, in __init__
    q_ini = tf.nn.embedding_lookup(params=self.dialogue_emb_weights, ids=tf.fill([bs], num_words))`

I changed num_words to num_words-1, and the problem was fixed, but then we had another problem:

`Exception has occurred: AlreadyExistsError
Resource __per_step_2/gradients/qgen/while/forward_pair/while/gru_cell/BiasAdd_1/Enter_grad/ArithmeticOptimizer/AddOpsRewrite_Add/tmp_var/N10tensorflow19TemporaryVariableOp6TmpVarE
	 [[{{node gradients/qgen/while/forward_pair/while/gru_cell/BiasAdd_1/Enter_grad/ArithmeticOptimizer/AddOpsRewrite_Add/tmp_var}}]]

During handling of the above exception, another exception occurred:

  File "/data/caishuo/ADVSE_v2/src/generic/tf_utils/evaluator.py", line 96, in execute
    return sess.run(output, feed_dict=feed_dict)
  File "/data/caishuo/ADVSE_v2/src/generic/tf_utils/evaluator.py", line 70, in process
    results = self.execute(sess, outputs, batch)
  File "/data/caishuo/ADVSE_v2/src/guesswhat/train/train_qgen_supervised.py", line 147, in <module>
    [train_loss, _] = evaluator.process(sess, train_iterator, outputs=outputs + [optimizer])`
   
This seems to happen when calculating the gradient, can you help me solve this problem or have a same problem,thanks
